operator/internal/manifests/storage/configure.go

```diff
@@ -294,3 +224,83 @@ func ensureCAForS3(p *corev1.PodSpec, tls *TLSConfig) corev1.PodSpec {
 		Volumes: volumes,
 	}
 }
+
+func wifEnabled(opts Options) bool {
+	storeType := opts.SharedStore
+	switch storeType {
+	case lokiv1.ObjectStorageSecretS3:
+		return opts.S3 != nil && opts.S3.STS
+	default:
+		return false
+	}
+}
+
+func setSATokenPath(opts *Options, osEnabled bool) {
+	var wiToken string
+	switch opts.SharedStore {
+	case lokiv1.ObjectStorageSecretS3:
+		wiToken = saTokenVolumeK8sDirectory
+		if osEnabled {
+			wiToken = saTokenVolumeOcpDirectory
+		}
+		opts.S3.WebIdentityTokenFile = wiToken
+	}
+}
+
+func saTokenVolumeMount(opts Options) corev1.VolumeMount {
+	var wiToken string
+	switch opts.SharedStore {
+	case lokiv1.ObjectStorageSecretS3:
+		wiToken = opts.S3.WebIdentityTokenFile
+	}
+	return corev1.VolumeMount{
+		Name:      saTokenVolumeName,
+		MountPath: wiToken,
+	}
+}
+
+func saTokenVolume(opts Options) corev1.Volume {
+	var audience string
+	storeType := opts.SharedStore
+	switch storeType {
+	case lokiv1.ObjectStorageSecretS3:
+		audience = opts.S3.Audience
+	}
```

Comment:
This is missing the default `audience` here. I believe in plain Kuberentes the audience is always `api`, but please double this suggestion before applying it.

Reply:
I was researching this and as far as I found `api` is only for Azure `audience: api://AzureADTokenExchange` like so. For aws I believe it would be `audience: sts.amazonaws.com`